code/game/machinery/droneDispenser.dm

```diff
@@ -51,28 +83,29 @@
 	..()
 	if(stat & (NOPOWER|BROKEN) || !anchored)
 		return
-	if(droneMadeRecently || metal < 1000 || glass < 1000)
+	if(use_materials && (metal < 1000 || glass < 1000))
+		return
+	if(droneMadeRecently)
+		icon_state = icon_recharging
 		return
 	droneMadeRecently = 1
-	var/turf/T = get_step(src,SOUTH)
 	visible_message("<span class='notice'>[src] whirs to life!</span>")
 	playsound(src, 'sound/items/rped.ogg', 50, 1)
-	icon_state = "make"
+	icon_state = icon_creating
 	spawn(30)
-		icon_state = "[initial(icon_state)]"
-		visible_message("<span class='notice'>[src] dispenses a drone shell.</span>")
-		metal -= 1000
-		glass -= 1000
-		if(metal < 0) metal = 0
-		if(glass < 0) glass = 0
-		if(istype(src, /obj/machinery/droneDispenser/syndrone))
-			new /obj/item/drone_shell/syndrone(T)
-		else
-			new /obj/item/drone_shell(T)
-		use_power(1000)
-		icon_state = "recharge"
+		icon_state = icon_on
+		if(use_materials)
+			metal -= 1000
+			glass -= 1000
+			if(metal < 0)
+				metal = 0
+			if(glass < 0)
+				glass = 0
+			use_power(1000)
```

Comment:
maybe add a ```power_cost``` variable, you've made everything else customisable, why the butts not this either.